frontend/src/App.tsx

```diff
@@ -1,18 +1,41 @@
-import React from 'react';
+import React, { useEffect, useState } from "react";
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+
 import "tabler-ui/dist/assets/css/dashboard.css";
-import './App.css';
-import Navbar from "./Navbar";
-import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { AuthService } from "./service/authService";
+import { Login } from "./pages/login";
+import { Homepage } from "./pages/homepage";
+import { PrivateRoute } from "./utils/privateRoute";
+import { PageNotFoundError } from "./pages/errorPage";
+
+export const authService = new AuthService();
 
 const App: React.FC = () => {
-    return (<Router>
-        <div className="App">
-            <Route path="/">
-            <Navbar/>
-            </Route>
-        </div>
-        </Router>
-    );
+    // Check the authentication then renders the page
+    let [checkedAuth, setCheckedAuth] = useState(false);
+    useEffect(() => {
+        authService.checkAuth().then(() => {
+            setCheckedAuth(true);
+        });
+    });
+
+    if (checkedAuth) {
+        return (
+            <Router>
+                {/* Switch imposes that only the first route to match the url will be rendered.
+                It avoids rendering PageNotFoundError on each page */}
+                <Switch>
+                    {/* A PrivateRoute is a route that requires to be authenticated to be accessible.
+                     If the user is a public one, they'll be redirected to the login page. */}
+                    <PrivateRoute exact path="/" component={Homepage} />
+                    <Route path="/login" component={Login} />
+                    <Route component={PageNotFoundError} />
+                </Switch>
+            </Router>
+        );
+    } else {
+        return <p>Loading...</p>;
```

Comment:
There is a <Spinner> component, might be interesting!

Reply:
`<Spinner>`*

Reply:
The user shouldn't even see it

Reply:
Then remove the else in that case…
Sorry, but this is not very convincing x)
UI is also about details